### PR TITLE
Improve multi framework resolution tests

### PR DESCRIPTION
**Pull request by vitek-karas**
Add some more tests for multi-framework resolution to cover certain specific corner cases

### CODE REVIEW

File: src/test/HostActivationTests/FrameworkResolution/RollForwardOnNoCandidateFxMultipleFrameworks.cs [hunk 1]
```diff
@@ -133,40 +133,47 @@ public MultipleFrameworks(SharedTestState sharedState)
...
                         .Version = versionReference),
-                resolvedFramework: resolvedFramework,
-                resultValidator: resolvedFramework != null ? null : new Action<CommandResult>(
-                    commandResult => commandResult.Should().Fail().And.FailedToSoftRollForward(MicrosoftNETCoreApp, "5.1.3", versionReference)));
+                resolvedFramework,
+                commandResult => commandResult.Should().Fail().And.FailedToSoftRollForward(MicrosoftNETCoreApp, "5.1.3", versionReference));
         }
 
-        // Soft roll forward from app's 5.1.1 (defaults) to inner framework reference with [specified version]
+        // Soft roll forward from inner framework reference [specified] to app's 6.0.1-preview.0 (defaults)
```
Comment (elinor-fung) [inline src/test/HostActivationTests/FrameworkResolution/RollForwardOnNoCandidateFxMultipleFrameworks.cs]: The app has `6.1.1-preview.0`, now?
